### PR TITLE
LPS-183744 Add Account Group On Site Initializer

### PR DESCRIPTION
**Pull request by joaodamasceno2001**

### CODE REVIEW

**Comment by joaodamasceno2001** [comment]
ci:test:sf

**Comment by joaodamasceno2001** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java [hunk 1]
```diff
@@ -356,6 +362,164 @@ public void testInitializeFromFile() throws Exception {
...
+		Assert.assertEquals("Test Account Group 1", accountGroup1.getName());
+
+		AdminAccountGroup accountGroup2 =
+			accountGroupResource.getAccountGroupByExternalReferenceCode(
+				"TESTACCOUNTGROUP2");
+
+		Assert.assertNotNull(accountGroup2);
+		Assert.assertEquals("Test Account Group 2", accountGroup2.getName());
+
+		AdminAccountGroup accountGroup3 =
```
[the remark this answers is not in the document]
Comment (nNilton) [reply modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java:388]: please apply the same logic for the other tests

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java [hunk 2]
```diff
@@ -356,6 +362,164 @@ public void testInitializeFromFile() throws Exception {
...
+				_serviceContext.fetchUser()
+			).build();
+
+		AdminAccountGroup accountGroup =
+			accountGroupResource.getAccountGroupByExternalReferenceCode(
+				"TESTACCOUNTGROUP1");
+
+		Assert.assertNotNull(account);
+
+		Assert.assertNotNull(accountGroup);
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java:454]: to test the assignments you dont need to check the account and account group
you can apply the same logic as _assertRolesAssignments()

**Review by nNilton — CHANGES_REQUESTED** [review]

**Comment by joaodamasceno2001** [comment]
ci:test:sf

**Comment by joaodamasceno2001** [comment]
ci:test:sf

**Comment by joaodamasceno2001** [comment]
ci:test:sf

**Comment by joaodamasceno2001** [comment]
ci:test:sf

**Comment by joaodamasceno2001** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 3]
```diff
@@ -639,6 +659,90 @@ protected void setServletContext(ServletContext servletContext) {
...
+			}
+
+			AccountGroup accountGroup =
+				_accountGroupLocalService.
+					fetchAccountGroupByExternalReferenceCode(
+						jsonObject.getString(
+							"accountGroupExternalReferenceCode"),
+						serviceContext.getCompanyId());
+
+			_accountGroupRelService.addAccountGroupRels(
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:705]: avoid NPE before using account group

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/build.gradle [hunk 4]
```diff
@@ -1,12 +1,14 @@
 dependencies {
 	testIntegrationCompile group: "com.liferay.portal", name: "com.liferay.portal.kernel", version: "default"
+	testIntegrationCompile project(":apps:account:account-api")
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/build.gradle:3]: please check if it's really necessary

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java [hunk 5]
```diff
@@ -355,6 +358,95 @@ public void testInitializeFromFile() throws Exception {
 		}
 	}
 
+	private void _assertAccountGroupAssign(
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java:361]: please rename to _assertAccountGroupAssignments

**Review by nNilton — CHANGES_REQUESTED** [review]

**Comment by joaodamasceno2001** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 6]
```diff
@@ -226,6 +234,10 @@
 public class BundleSiteInitializer implements SiteInitializer {
 
 	public BundleSiteInitializer(
+		AccountEntryLocalService accountEntryLocalService,
+		AccountGroupLocalService accountGroupLocalService,
+		AdminAccountGroupResource.Factory adminAccountGroupResourceFactory,
+		AccountGroupRelService accountGroupRelService,
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:240]: sort

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 7]
```diff
@@ -298,6 +310,10 @@ public BundleSiteInitializer(
 		WorkflowDefinitionLinkLocalService workflowDefinitionLinkLocalService,
 		WorkflowDefinitionResource.Factory workflowDefinitionResourceFactory) {
 
+		_accountEntryLocalService = accountEntryLocalService;
+		_accountGroupLocalService = accountGroupLocalService;
+		_adminAccountGroupResourceFactory = adminAccountGroupResourceFactory;
+		_accountGroupRelService = accountGroupRelService;
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:316]: sort

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 8]
```diff
@@ -452,8 +468,12 @@ public void initialize(long groupId) throws InitializationException {
 				siteNavigationMenuItemSettingsBuilder =
 					new SiteNavigationMenuItemSettingsBuilder();
 
+			_invoke(() -> _addAccountGroups(serviceContext));
+
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:472]: try to remove black line

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerExtender.java [hunk 9]
```diff
@@ -131,6 +135,8 @@ public SiteInitializerExtension addingBundle(
 
 		SiteInitializerExtension siteInitializerExtension =
 			new SiteInitializerExtension(
+				_accountEntryLocalService, _accountGroupLocalService,
+				_adminAccountGroupResourcefactory, _accountGroupRelService,
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerExtender.java:139]: sort

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerExtender.java [hunk 10]
```diff
@@ -243,6 +249,8 @@ private void _addFile(File file) throws Exception {
...
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerExtender.java:253]: sort

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerExtension.java [hunk 11]
```diff
@@ -96,6 +100,10 @@
 public class SiteInitializerExtension {
 
 	public SiteInitializerExtension(
+		AccountEntryLocalService accountEntryLocalService,
+		AccountGroupLocalService accountGroupLocalService,
+		AdminAccountGroupResource.Factory adminAccountGroupResourceFactory,
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerExtension.java:105]: sort

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerFactoryImpl.java [hunk 12]
```diff
@@ -120,6 +124,8 @@ public SiteInitializer create(File file, String symbolicName)
 			null);
 
 		BundleSiteInitializer bundleSiteInitializer = new BundleSiteInitializer(
+			_accountEntryLocalService, _accountGroupLocalService,
+			_adminAccountGroupResourceFactory, _accountGroupRelService,
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerFactoryImpl.java:128]: sort

**Review by nNilton — CHANGES_REQUESTED** [review]

**Pull request closed without merging**